daemon/src/monitor.rs

```diff
@@ -102,14 +137,17 @@ impl Cfd {
         use CfdEvent::*;
         match event.event {
             ContractSetupCompleted { dlc, .. } => Self {
-                params: Some(MonitorParams::new(dlc)),
+                params: Some(MonitorParams::new(dlc.clone())),
                 monitor_lock_finality: true,
                 monitor_commit_finality: true,
                 monitor_cet_timelock: true,
                 monitor_refund_timelock: true,
                 monitor_refund_finality: true,
                 monitor_revoked_commit_transactions: false,
                 monitor_collaborative_settlement_finality: None,
+                lock_tx: Some(dlc.lock.0),
+                cet: self.cet,
```

Comment:
Particularly for `ContractSetupCompleted` I find this a bit confusing. Upon contract setup completed I would expect `cet` and `commit_tx` to be set to `None`. I think that is what effectively happens here, because it *will* always be `None` at this point. 

Is there any specific reason why we pass through the value here?

Reply:
nope, I am not super confident with these state transitions so I copied what we were doing in an other file. will fix